src/scratch.h

```diff
@@ -12,28 +12,29 @@
 /* The typedef is used internally; the struct name is used in the public API
  * (where it is exposed as a different typedef) */
 typedef struct secp256k1_scratch_space_struct {
-    void *data[SECP256K1_SCRATCH_MAX_FRAMES];
+    unsigned char magic[8];
```

Comment:
Probably not this PR but what do people think about randomizing this? Then it could protect not only against accidental misuse but also against stack smashing etc. This is just a random thought, I haven't really thought about whether it makes sense or not.

Reply:
Not this PR :)